### PR TITLE
Basic implementation of upnp search

### PR DESCRIPTION
**Pull request by astylos**
Implements some of the functionality defined in standardizeddcps\MediaServer_4 and MediaRenderer_3\UPnP-av-ContentDirectory-v4-Service-20150630.pdf
Document available in upnpresources.zip, download from here:
https://openconnectivity.org/developer/specifications/upnp-resources/upnp

Example search sent by a client such as foobar2000 or BubbleUPNP:
```
upnp:class derivedfrom "object.item.audioItem" and (upnp:album contains "Midnight")
```
Main changes to existing codebase:
1. New mt_metadata table
2. Schema upgrade on gerbera startup includes migration of existing metadata from mt_cds_object to mt_metadata (but it **does not** remove metadata from mt_cds_object)
3. All new metadata stored in mt_metadata

See #249 for more background/issues that came up when working on this.

Most obvious functional issue is albums cannot be found since no album metadata stored - see #255 Album metadata not persisted.

Regarding the code itself, I tried to use modern C++ and avoid zmm in new classes. At times I did wonder whether that was a good idea given how unique_ptr/shared_ptr work so differently to zmm::Ref.

### CODE REVIEW

**Comment by v00d00** [comment]
@elmodaddyb does this conflict with your gtest changes?

**Comment by astylos** [comment]
I merged those changes onto my branch so shouldn't be any issue.

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -7,4 +7,5 @@ get_target_property(GERBERA_INTERFACE_LIBRARIES gerbera INTERFACE_LINK_LIBRARIES
 
 ## Add tests below
 add_subdirectory(test_dictionary)
-add_subdirectory(test_runtime)
\ No newline at end of file
+add_subdirectory(test_runtime)
+add_subdirectory(test_searchhandler)
```
Comment (elmodaddyb) [inline test/CMakeLists.txt]: :100:

**Comment by elmodaddyb** [comment]
TDD for the win!!!  Hopefully the testing model is repeatable and easy going forward.  So far I've written 3 suites, 2 using fixures, 1 with mocks.  The CMAKE layout seems to work and the `libgerbera` separation is well suited.

**Comment by v00d00** [comment]
@astylos Are you happy for me to squash merge this, or would you rather tidy the history up yourself?

**Comment by astylos** [comment]
@v00d00 Yes, go ahead and squash merge

**Comment by v00d00** [comment]
@astylos Thanks!

**Comment by compguy284** [comment]
I recently installed gerbera and have been having problems playing content on my PS3. First try it says cannot play content, the second try succeeds. I have narrowed the cause down to this commit with `mark-played-items enabled="yes"` since d04736dc73dae3c78968029845363fa65f0c3d51 works just fine.

I didn't know if I should do a bug report or if commenting here is/was good enough.

**Comment by elmodaddyb** [comment]
Hello @compguy284 - I looked at the code and the `marked-played-items` prepends `yes` or `no` to the title text based on whether the item has been played.

I have a PS3 so I will run a few tests to see how it works.

**Comment by compguy284** [comment]
I submitted a pull request that fixes the problem for me.

**Pull request merged**